cirq/circuits/circuit_dag.py

```diff
@@ -175,8 +175,51 @@ def get_next_node(succ: networkx.classes.coreviews.AtlasView
     def all_operations(self) -> Iterator[ops.Operation]:
         return (node.val for node in self.ordered_nodes())
 
+    def all_qubits(self):
+        return frozenset(q for node in self.nodes for q in node.val.qubits)
+
     def to_circuit(self) -> circuit.Circuit:
         return circuit.Circuit.from_ops(
                     self.all_operations(),
                     strategy=circuit.InsertStrategy.EARLIEST,
                     device=self.device)
+
+    def is_topologically_sorted(self, operations: ops.OP_TREE) -> bool:
```

Comment:
Oh, the operations are supposed to be the same as the ones in the circuit? Maybe we should an exception if there's some missing or extra ones?

Reply:
Yes, that is the intention.

If a user passes in an unexpected operation, do you expect that they did so as a mistake or do you expect that they intended to use this functionality?

Reply:
I'm not sure. I'm using this to test code that takes a `CircuitDag` and a `Device` and outputs a `Schedule` including swaps, so I'm only interested in whether it's exactly right or not.

Having to distinguish between "bad order" and "different set of operations" would complicate the logic of the function, so I'm tempted to not do that. Assuming both are treated the same way, I'd rather peacefully return `False` and let the caller deal with what to do next, rather than always throw an error. As a side note, I wish the device validation methods did the same; in many case I want to know whether or not an operation is valid rather than assert that it is.

Reply:
@Strilanc ping

Reply:
I'm still not convinced that we want to add this. I'd prefer to have it in contrib.

@dabacon what is your opinion?